src/lib/linux-api/src/signal.rs

```diff
@@ -0,0 +1,1192 @@
+        lhs: *const linux_sigset_t,
+        rhs: *const linux_sigset_t,
+    ) -> linux_sigset_t {
+        let lhs = unsafe { lhs.as_ref().unwrap() };
+        let rhs = unsafe { rhs.as_ref().unwrap() };
+        SigSet(*lhs & *rhs).0
+    }
+
+    #[no_mangle]
+    pub unsafe extern "C" fn linux_signotset(set: *const linux_sigset_t) -> linux_sigset_t {
+        let set = unsafe { set.as_ref().unwrap() };
+        SigSet(!*set).0
+    }
+
+    #[no_mangle]
+    pub unsafe extern "C" fn linux_siglowest(set: *const linux_sigset_t) -> i32 {
+        let set = SigSet::deref_c(unsafe { set.as_ref().unwrap() });
+        match set.lowest() {
+            Some(s) => s.into(),
+            None => 0,
+        }
+    }
+
+    #[no_mangle]
+    pub extern "C" fn linux_defaultAction(signo: i32) -> LinuxDefaultAction {
+        let sig = Signal::try_from(signo).unwrap();
+        defaultaction(sig)
+    }
+
+    #[no_mangle]
+    pub extern "C" fn linux_siginfo_new_for_kill(
+        lsi_signo: i32,
+        sender_pid: i32,
+        sender_uid: u32,
+    ) -> linux_siginfo_t {
+        let signal = Signal::try_from(lsi_signo).unwrap();
+        SigInfo::peel(SigInfo::new_for_kill(signal, sender_pid, sender_uid))
+    }
+
+    #[no_mangle]
+    pub extern "C" fn linux_siginfo_new_for_tkill(
+        lsi_signo: i32,
+        sender_pid: i32,
+        sender_uid: u32,
+    ) -> linux_siginfo_t {
+        let signal = Signal::try_from(lsi_signo).unwrap();
+        SigInfo::peel(SigInfo::new_for_tkill(signal, sender_pid, sender_uid))
+    }
+
+    /// Returns the handler if there is one, else NULL.
+    // Ideally we'd want to return Option<LinuxSigHandler> here, but cbindgen doesn't
+    // decay it back to LinuxSigHandler.
+    #[no_mangle]
+    #[allow(clippy::transmute_null_to_fn)]
+    pub unsafe extern "C" fn linux_sigaction_handler(
+        sa: *const linux_sigaction,
+    ) -> LinuxSigHandlerFn {
+        let sa = SigAction::wrap_ref(unsafe { sa.as_ref().unwrap() });
+        match sa.handler() {
+            SignalHandler::Handler(h) => h,
+            _ => unsafe { core::mem::transmute(core::ptr::null::<core::ffi::c_void>()) },
+        }
+    }
+
+    /// Returns the action if there is one, else NULL.
+    // Ideally we'd want to return Option<LinuxSigHandler> here, but cbindgen doesn't
+    // decay it back to LinuxSigHandler.
+    #[no_mangle]
+    #[allow(clippy::transmute_null_to_fn)]
+    pub unsafe extern "C" fn linux_sigaction_action(
+        sa: *const linux_sigaction,
+    ) -> LinuxSigActionFn {
+        let sa = SigAction::wrap_ref(unsafe { sa.as_ref().unwrap() });
+        match sa.handler() {
+            SignalHandler::Action(h) =>
+            // We need to transmute the function pointer here, to one that takes
+            // the exported C types.
+            unsafe { core::mem::transmute::<SignalActionFn, LinuxSigActionFn>(h) },
+            _ => unsafe { core::mem::transmute(core::ptr::null::<core::ffi::c_void>()) },
+        }
```

Comment:
Same as above transmute comments (about making the first transmute explicit, and the second transmute is UB).